Tools/YamlCreate.ps1

```diff
@@ -31,232 +31,663 @@ filter TrimString {
+        Write-Host -ForegroundColor 'DarkCyan' -Object ' Update Manifest'
+    Write-Host -ForegroundColor 'DarkCyan' -NoNewline "`n["; Write-Host -NoNewline '3'; Write-Host -ForegroundColor DarkCyan -NoNewline "]"; `
+        Write-Host -ForegroundColor 'DarkCyan' -Object ' New Locale'
+    Write-Host -ForegroundColor 'DarkCyan' -NoNewline "`n["; Write-Host -NoNewline 'q'; Write-Host -ForegroundColor DarkCyan -NoNewline "]"; `
+        Write-Host -ForegroundColor 'Red' -Object ' Any key to quit'
+    $OptionMenu = Read-Host "`nSelection"
+    switch ($OptionMenu) {
+        '1' {$Option = 'New'}
+        '2' {$Option = 'Update'}
+        '3' {$Option = 'NewLocale'}
+        default {exit}
+    }
 }
 
-Write-Host "Url: $URL"
-Write-Host "Sha256: $Hash"
+##########################################
+#region checksum
 
-Write-Host $NewLine
-Write-Host "File downloaded. Please Fill out required fields."
 
 #endregion
 ##########################################
 
 ##########################################
 #region Read in metadata
 
-while ($ID.Length -lt 4 -or $ID.Length -ge 255) {
-    Write-Host 'Enter the package Id, in the following format <Publisher.Appname>'
-    $ID = Read-Host -Prompt 'For example: Microsoft.Excel' | TrimString
-}
-
-$host.UI.RawUI.ForegroundColor = "White"
 while ([string]::IsNullOrWhiteSpace($Publisher) -or $Publisher.Length -ge 128) {
-    $Publisher = Read-Host -Prompt 'Enter the publisher' | TrimString
+    Write-Host ''
+    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the publisher. For example: Microsoft'
+    $Publisher = Read-Host -Prompt 'Publisher' | TrimString
 }
 
-while ([string]::IsNullOrWhiteSpace($AppName) -or $AppName.Length -ge 128) {
-    $AppName = Read-Host -Prompt 'Enter the application name' | TrimString
-    $AppNameFolder = $AppName -replace '\s',''
+while ([string]::IsNullOrWhiteSpace($PackageName) -or $PackageName.Length -ge 128) {
+    Write-Host ''
+    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the application name. For example: Teams'
+    $PackageName = Read-Host -Prompt 'Application Name' | TrimString
 }
 
-while ([string]::IsNullOrWhiteSpace($version)) {
-    $version = Read-Host -Prompt 'Enter the version. For example: 1.0.0, 1.0.0.0, 1.0' | TrimString
-    $ManifestName = $version + ".yaml"
+while ([string]::IsNullOrWhiteSpace($PackageVersion) -or $PackageName.Length -ge 128) {
+    Write-Host ''
+    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the version. for example: 1.33.7'
+    $PackageVersion = Read-Host -Prompt 'Version' | TrimString
 }
 
-while ([string]::IsNullOrWhiteSpace($License) -or $License.Length -ge 40) {
-    $License = Read-Host -Prompt 'Enter the License, For example: MIT, or Copyright (c) Microsoft Corporation' | TrimString
+$PackageIdentifier = [System.String]::Concat($Publisher.Replace(' ',''),'.',$PackageName.Replace(' ',''))
```

Comment:
We shouldn't assume that the user wants the simple concatenation of these two as their packageId, there are plenty of cases where that's not true, for example https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/VisualStudioCodeInsiders/System-x64/1.55.9.b353f06339/Microsoft.VisualStudioCodeInsiders.System-x64.yaml.

Reply:
As with the code i posted above i now have a issue where i need to spilt the Publisher from the PackageName when there is 2 dots i the PackageName

Before i saw the Manifest you posted i made this string
```
$PackageIdentifierPub = $PackageIdentifier.Split('.') | Select -First '1'
$PackageIdentifierApp = $PackageIdentifier.Split('.') | Select -Skip '1'
```
However i don't see a easy way around this other than to do Foreach dot after the first skip

Reply:
I'm not sure I totally get what you're saying. I believe you could just do:
`
$PackageIdentifierPub = $PackageIdentifier.Split('.')[0]
$PackageIdentifierApp = $PackageIdentifier.Split('.')[0..($PackageIdentifier.Count-1)] | String-Join "."
`
Not very pretty I know

Btw, it looks like you're currently using $PackageIdentifierPub on line 95 without first defining it